vxportia/tests/test_protocol.py

```diff
@@ -0,0 +1,141 @@
+import pkg_resources
+import json
+
+from twisted.trial.unittest import TestCase
+from twisted.internet import reactor
+from twisted.internet.defer import inlineCallbacks, Deferred
+from twisted.internet.protocol import Factory
+from twisted.internet.task import Clock
+from twisted.test.proto_helpers import StringTransportWithDisconnection
+
+from vxportia.protocol import PortiaProtocol, PortiaProtocolException
+
+from portia.portia import Portia
+from portia.utils import (
+    start_redis, compile_network_prefix_mappings)
+
+
+class TestPortiaProtocol(TestCase):
+
+    timeout = 1
+
+    @inlineCallbacks
+    def setUp(self):
+        self.redis = yield start_redis()
+        self.addCleanup(self.redis.disconnect)
+
+        self.portia = Portia(
+            self.redis,
+            network_prefix_mapping=compile_network_prefix_mappings(
+                [pkg_resources.resource_filename(
+                    'portia', 'assets/mappings/*.mapping.json')]))
+        self.addCleanup(self.portia.flush)
+
+        factory = Factory.forProtocol(PortiaProtocol)
+        self.proto = factory.buildProtocol(None)
+        self.proto.clock = Clock()
+        self.transport = StringTransportWithDisconnection()
+        self.proto.makeConnection(self.transport)
+
+    def read_command(self):
+        response_d = Deferred()
+
+        def check(d):
+            val = self.transport.value()
+            self.transport.clear()
+            if val:
+                d.callback(json.loads(val))
+                return
+            reactor.callLater(0, check, d)
+        reactor.callLater(0, check, response_d)
+
+        return response_d
+
+    def reply(self, command, response={}, status='ok', message=None):
+        self.proto.parseLine(json.dumps({
```

Comment:
Should be not feed this into the transport rather and have that call the proto? Or failing that at least call `lineReceived` or `dataReceived`?